### PR TITLE
Updates for conditional expression type breaking change

### PR DESCRIPTION
**Pull request by cston**
See https://github.com/dotnet/roslyn/pull/50755

### CODE REVIEW

**Comment by cston** [comment]
cc @jaredpar

File: src/libraries/System.Data.Common/src/System/Xml/DataSetMappper.cs [hunk 1]
```diff
@@ -314,13 +314,13 @@ internal bool IsRegionRadical(XmlBoundElement rowElem)
                 }
 
                 // must have no attributes
-                if (e.HasAttributes)
+                if (e!.HasAttributes)
```
Comment (stephentoub) [inline src/libraries/System.Data.Common/src/System/Xml/DataSetMappper.cs:317]: Why wasn't this failing before?

**Review by stephentoub — APPROVED** [review]

File: src/coreclr/tools/Common/JitInterface/CorInfoBase.cs [hunk 2]
```diff
@@ -61,7 +61,7 @@ static byte _getMethodInfo(IntPtr thisHandle, IntPtr* ppException, CORINFO_METHO
             var _this = GetThis(thisHandle);
             try
             {
-                return _this.getMethodInfo(ftn, info) ? 1 : 0;
+                return _this.getMethodInfo(ftn, info) ? (byte)1 : (byte)0;
```
Comment (jkotas) [inline src/coreclr/tools/Common/JitInterface/CorInfoBase.cs:64]: This is auto-generated file. Could you please also update the auto-generator (path to the auto-generator is at the top of the file)?

File: src/libraries/System.Data.Common/src/System/Xml/DataSetMappper.cs (hunk 1 above)
Comment (cston) [reply src/libraries/System.Data.Common/src/System/Xml/DataSetMappper.cs:317]: It looks like the compiler used to infer `e != null` based on the call to `GetColumnSchemaForNode(rowElem, e!)`. Now, the compiler will only infer `e != null` when the warning that should be reported passing `e` to `GetColumnSchemaForNode(rowElem, e)` is not suppressed. See [sharplab.io](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAxAOwK4QgQwEYQFMACA1PQgWACgABAJiJoEYB2agb2qO8aYDZGAFiIBZJgAoA9rgBWBAMbAA/EUkBKLj05Ueu3gAYVAZyIBeVUWwnm+gNya9ouuKMBCNfZ2OiAfTNEjADoAGVIAc2AAC09dAF8HXgEaYRFnaTlFVQ0vbm046ligA).

**Comment by ViktorHofer** [comment]
Is this PR ready? The next Arcade dependency update will come with the updated compiler that throws these warnings: https://dev.azure.com/dnceng/internal/_build/results?buildId=976890&view=results

**Pull request merged**